### DISCUSSION TITLE
Tracks get terminated

### DISCUSSION THREAD

**Discussion by dvorel** [trackmania-rl/tmrl#60] [Q&A] [answered]
Hi, every track except tmrl-test gets terminated after 81 steps. I have tried changing "ep_max_length" in config, but without luck.

Thanks in advance.

**Comment by yannbouteiller** [comment, marked as answer]
Hi, this is because you need to record a reward function for your track, the default reward function is for tmrl-test/train.

You can find the instructions for doing this [here](https://github.com/trackmania-rl/tmrl/blob/master/readme/get_started.md#train-your-own-self-driving-ai)

Happy driving :)

**Reply by dvorel** [reply]
Thank you! Now it works, I didn't read it thoroughly..

**Reply by dvorel** [reply]
Is there any way to speed up game? I found tminterface but unfortunately it doesn't support tm 2020

I'm asking here so I don't open new thread..

**Reply by yannbouteiller** [reply]
I don't think there is a way to speed up the game with TM20. But speeding up the game would most likely be contrary to Real-Time Reinforcement Learning, as inference and observation capture durations would not be correctly represented anymore unless you get into advanced tricks (especially for observation capture).

`tmrl` allows you to collect data from several workers in parallel, though.
